src/generator.cs

```diff
@@ -2609,42 +2605,27 @@ void GenerateTrampolinesForQueue (TrampolineInfo [] queue)
 			// Now generate the class that allows us to invoke a Objective-C block from C#
 			//
 			print ("");
-			print ("internal class {0} {{", ti.NativeInvokerName);
+			print ("internal class {0} : TrampolineBlockBase {{", ti.NativeInvokerName);
```

Comment:
Can/should we seal the class?

Reply:
Yes! That would help for XM. For XI the linker will, by default, seal types that are not subclassed.